createFeatuerSetFromImageSet may throw java.lang.UnsatisfiedLinkError: org.opencv.core.Mat.n_Mat()J

Code snippets:
```
image_set = ImageSet.read(image_path,sc=sc, with_label=True, one_based_label=False)
        train_transformer = ChainedPreprocessing([ImageBytesToMat(),
                                                  ImageResize(256, 256),
                                                  ImageRandomCrop(224, 224),
                                                  ImageRandomPreprocessing(ImageHFlip(), 0.5),
                                                  ImageChannelNormalize(
                                                          0.485, 0.456, 0.406,
                                                          0.229, 0.224, 0.225),
                                                  ImageMatToTensor(to_RGB=True, format="NHWC"),
                                                  ImageSetToSample(input_keys=["imageTensor"],
                                                                       target_keys=["label"])
                                                      ])
        feature_set = FeatureSet.image_frame(image_set.to_image_frame())
        feature_set = feature_set.transform(train_transformer)
        feature_set = feature_set.transform(ImageFeatureToSample())
```
Error message:
```
Driver stacktrace:
	at org.apache.spark.scheduler.DAGScheduler.org$apache$spark$scheduler$DAGScheduler$$failJobAndIndependentStages(DAGScheduler.scala:1889)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1877)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1876)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:1876)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:926)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:926)
	at scala.Option.foreach(Option.scala:257)
	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:926)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2110)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2059)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2048)
	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:737)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2061)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2082)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2101)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2126)
	at org.apache.spark.rdd.RDD.count(RDD.scala:1168)
	at com.intel.analytics.zoo.feature.DistributedFeatureSet$$anon$1.cache(FeatureSet.scala:139)
	at com.intel.analytics.zoo.feature.DistributedDataSetWrapper.cache(FeatureSet.scala:216)
	at com.intel.analytics.bigdl.optim.AbstractOptimizer.prepareInput(AbstractOptimizer.scala:281)
	at com.intel.analytics.bigdl.optim.DistriOptimizer.prepareInput(DistriOptimizer.scala:809)
	at com.intel.analytics.zoo.pipeline.api.keras.models.InternalDistriOptimizer.train(Topology.scala:1200)
	at com.intel.analytics.zoo.pipeline.api.keras.models.InternalDistriOptimizer.train(Topology.scala:1472)
	at com.intel.analytics.zoo.pipeline.api.keras.models.InternalDistriOptimizer.train(Topology.scala:1145)
	at com.intel.analytics.zoo.pipeline.estimator.Estimator.train(Estimator.scala:190)
	at com.intel.analytics.zoo.pipeline.estimator.python.PythonEstimator.estimatorTrainMiniBatch(PythonEstimator.scala:113)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.InvalidClassException: com.intel.analytics.bigdl.transform.vision.image.opencv.OpenCVMat; unable to create instance
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:1788)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1351)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:371)
	at scala.collection.mutable.HashMap$$anonfun$readObject$1.apply(HashMap.scala:174)
	at scala.collection.mutable.HashMap$$anonfun$readObject$1.apply(HashMap.scala:174)
	at scala.collection.mutable.HashTable$class.init(HashTable.scala:109)
	at scala.collection.mutable.HashMap.init(HashMap.scala:40)
	at scala.collection.mutable.HashMap.readObject(HashMap.scala:174)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at java.io.ObjectStreamClass.invokeReadObject(ObjectStreamClass.java:1058)
	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:1900)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:1801)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1351)
	at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2000)
	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:1924)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:1801)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1351)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:371)
	at org.apache.spark.serializer.JavaDeserializationStream.readObject(JavaSerializer.scala:75)
	at org.apache.spark.serializer.DeserializationStream.readValue(Serializer.scala:158)
	at org.apache.spark.serializer.DeserializationStream$$anon$2.getNext(Serializer.scala:188)
	at org.apache.spark.serializer.DeserializationStream$$anon$2.getNext(Serializer.scala:185)
	at org.apache.spark.util.NextIterator.hasNext(NextIterator.scala:73)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:439)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:409)
	at org.apache.spark.util.CompletionIterator.hasNext(CompletionIterator.scala:31)
	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:439)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:409)
	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
	at scala.collection.generic.Growable$class.$plus$plus$eq(Growable.scala:59)
	at scala.collection.mutable.ArrayBuffer.$plus$plus$eq(ArrayBuffer.scala:104)
	at scala.collection.mutable.ArrayBuffer.$plus$plus$eq(ArrayBuffer.scala:48)
	at scala.collection.TraversableOnce$class.to(TraversableOnce.scala:310)
	at scala.collection.AbstractIterator.to(Iterator.scala:1334)
	at scala.collection.TraversableOnce$class.toBuffer(TraversableOnce.scala:302)
	at scala.collection.AbstractIterator.toBuffer(Iterator.scala:1334)
	at scala.collection.TraversableOnce$class.toArray(TraversableOnce.scala:289)
	at scala.collection.AbstractIterator.toArray(Iterator.scala:1334)
	at com.intel.analytics.zoo.feature.DRAMFeatureSet$$anonfun$7.apply(FeatureSet.scala:637)
	at com.intel.analytics.zoo.feature.DRAMFeatureSet$$anonfun$7.apply(FeatureSet.scala:636)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitions$1$$anonfun$apply$23.apply(RDD.scala:801)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitions$1$$anonfun$apply$23.apply(RDD.scala:801)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD$$anonfun$7.apply(RDD.scala:337)
	at org.apache.spark.rdd.RDD$$anonfun$7.apply(RDD.scala:335)
	at org.apache.spark.storage.BlockManager$$anonfun$doPutIterator$1.apply(BlockManager.scala:1165)
	at org.apache.spark.storage.BlockManager$$anonfun$doPutIterator$1.apply(BlockManager.scala:1156)
	at org.apache.spark.storage.BlockManager.doPut(BlockManager.scala:1091)
	at org.apache.spark.storage.BlockManager.doPutIterator(BlockManager.scala:1156)
	at org.apache.spark.storage.BlockManager.getOrElseUpdate(BlockManager.scala:882)
	at org.apache.spark.rdd.RDD.getOrCompute(RDD.scala:335)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:286)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD$$anonfun$7.apply(RDD.scala:337)
	at org.apache.spark.rdd.RDD$$anonfun$7.apply(RDD.scala:335)
	at org.apache.spark.storage.BlockManager$$anonfun$doPutIterator$1.apply(BlockManager.scala:1165)
	at org.apache.spark.storage.BlockManager$$anonfun$doPutIterator$1.apply(BlockManager.scala:1156)
	at org.apache.spark.storage.BlockManager.doPut(BlockManager.scala:1091)
	at org.apache.spark.storage.BlockManager.doPutIterator(BlockManager.scala:1156)
	at org.apache.spark.storage.BlockManager.getOrElseUpdate(BlockManager.scala:882)
	at org.apache.spark.rdd.RDD.getOrCompute(RDD.scala:335)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:286)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:121)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:408)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:414)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	... 1 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.GeneratedSerializationConstructorAccessor90.newInstance(Unknown Source)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at java.io.ObjectStreamClass.newInstance(ObjectStreamClass.java:1006)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:1785)
	... 76 more
Caused by: java.lang.UnsatisfiedLinkError: org.opencv.core.Mat.n_Mat()J
	at org.opencv.core.Mat.n_Mat(Native Method)
	at org.opencv.core.Mat.<init>(Mat.java:24)
	... 80 more
```

The image path is a folder contains 2200 jpg images, but the ImageSet.read only get one partition, the reading will apply a ImageBytesToMat by default, so only the executor who read the partition has load  OPENCV, then `feature_set = FeatureSet.image_frame(image_set.to_image_frame())` will repartition the RDD[ImageFrame], but other nodes will return a failure during the deseraization, as they don't load OpenCV before the deseralization.

--- a/zoo/src/main/resources/spark-analytics-zoo.conf
+++ b/zoo/src/main/resources/spark-analytics-zoo.conf
@@ -30,6 +30,6 @@
 spark.shuffle.reduceLocality.enabled                false
 spark.shuffle.blockTransferService                  nio
 spark.scheduler.minRegisteredResourcesRatio         1.0
-spark.scheduler.maxRegisteredResourcesWaitingTime   100s
+spark.scheduler.maxRegisteredResourcesWaitingTime   3600s
 spark.speculation                                   false
 spark.serializer                                    org.apache.spark.serializer.JavaSerializer

--- a/zoo/src/main/scala/com/intel/analytics/zoo/feature/python/PythonFeatureSet.scala
+++ b/zoo/src/main/scala/com/intel/analytics/zoo/feature/python/PythonFeatureSet.scala
@@ -16,10 +16,9 @@
 
 package com.intel.analytics.zoo.feature.python
 
-import java.util
-
 import com.intel.analytics.bigdl.DataSet
 import com.intel.analytics.bigdl.dataset.{MiniBatch, Transformer, Sample => JSample}
+import com.intel.analytics.bigdl.opencv.OpenCV
 import com.intel.analytics.bigdl.python.api.Sample
 import com.intel.analytics.bigdl.tensor.Tensor
 import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
@@ -29,7 +28,9 @@ import com.intel.analytics.zoo.common.PythonZoo
 import com.intel.analytics.zoo.feature.FeatureSet
 import com.intel.analytics.zoo.feature.pmem.MemoryType
 import com.intel.analytics.zoo.pipeline.api.keras.layers.utils.EngineRef
+import org.apache.spark.SparkContext
 import org.apache.spark.api.java.JavaRDD
+import org.apache.spark.storage.StorageLevel
 
 import scala.collection.JavaConverters._
 import scala.reflect.ClassTag
@@ -39,15 +40,33 @@ object PythonFeatureSet {
   def ofFloat(): PythonFeatureSet[Float] = new PythonFeatureSet[Float]()
 
   def ofDouble(): PythonFeatureSet[Double] = new PythonFeatureSet[Double]()
+
+  protected def loadOpenCv(sc: SparkContext): Unit = {
+    val nodeNumber = EngineRef.getNodeNumber()
+    val loadOpenCvRdd = sc.parallelize(
+      Array.tabulate(nodeNumber)(_ => "dummy123123"), nodeNumber * 10)
+      .mapPartitions(_ => (0 until 20000000).toIterator)
+      .coalesce(nodeNumber)
+      .setName("LoadOpenCvRdd")
+      .persist(StorageLevel.DISK_ONLY)
+    loadOpenCvRdd.count()
+    loadOpenCvRdd.mapPartitions{v =>
+      OpenCV.isOpenCVLoaded()
+      v
+    }.count()
+    loadOpenCvRdd.unpersist()
+  }
 }
 
 @SerialVersionUID(7610684191490849169L)
 class PythonFeatureSet[T: ClassTag](implicit ev: TensorNumeric[T]) extends PythonZoo[T] {
+  import PythonFeatureSet.loadOpenCv
   def createFeatureSetFromImageFrame(
         imageFrame: ImageFrame,
         memoryType: String,
         sequentialOrder: Boolean, shuffle: Boolean): FeatureSet[ImageFeature] = {
     require(imageFrame.isDistributed(), "Only support distributed ImageFrame")
+    loadOpenCv(imageFrame.toDistributed().rdd.sparkContext)
     FeatureSet.rdd(imageFrame.toDistributed().rdd, MemoryType.fromString(memoryType),
       sequentialOrder = sequentialOrder, shuffle = shuffle)
   }

--- a/zoo/src/main/scala/com/intel/analytics/zoo/feature/python/PythonFeatureSet.scala
+++ b/zoo/src/main/scala/com/intel/analytics/zoo/feature/python/PythonFeatureSet.scala
@@ -45,12 +45,11 @@ object PythonFeatureSet {
     val nodeNumber = EngineRef.getNodeNumber()
     val loadOpenCvRdd = sc.parallelize(
       Array.tabulate(nodeNumber)(_ => "dummy123123"), nodeNumber * 10)
-      .mapPartitions(_ => (0 until 20000000).toIterator)
+      .mapPartitions(_ => (0 until 2000000).toIterator)
       .coalesce(nodeNumber)
       .setName("LoadOpenCvRdd")
-      .persist(StorageLevel.DISK_ONLY)
     loadOpenCvRdd.count()
-    loadOpenCvRdd.mapPartitions{v =>
+    loadOpenCvRdd.coalesce(nodeNumber).mapPartitions{v =>
       OpenCV.isOpenCVLoaded()
       v
     }.count()

